beacon-chain/core/epoch/precompute/reward_penalty.go

```diff
@@ -163,7 +163,7 @@ func ProposersDelta(state *stateTrie.BeaconState, pBal *Balance, vp []*Validator
 	baseRewardsPerEpoch := params.BeaconConfig().BaseRewardsPerEpoch
 	proposerRewardQuotient := params.BeaconConfig().ProposerRewardQuotient
 	for _, v := range vp {
-		if v.ProposerIndex >= uint64(len(rewards)) {
+		if int(v.ProposerIndex) >= len(rewards) {
```

Comment:
this is downcasting, seems like a big bug if it goes through and we exceed length of max int